### ISSUE TITLE
 Dependencies between detector runs on LBP

### ISSUE THREAD

**Issue by almln** [Itseez/opencv#5442]
There is some bug with cv::CascadeClassifier in case of lbpcascade_frontalface:
We have 3 photos (attached files arranged as 02463d131, 02463d139, 02463d142 respectively). Сonsistently run detector on them. For the last picture we get the wrong result. Remove, for example, the second photo. Now, the result for the third picture changed and became correct.
This observation is true only for lbp cascades. On haarcascade all is ok.
See attached files for more info.

Build info:
openCV 3.0.0, vc12, static

Code:
```cpp
#include <opencv2\opencv.hpp>

int main() {

    std::string f1 = "C:/test/detection/imgs/02463d131.jpg";
    std::string f2 = "C:/test/detection/imgs/02463d139.jpg";
    std::string f3 = "C:/test/detection/imgs/02463d142.jpg";

    cv::Mat im1 = cv::imread(f1, CV_LOAD_IMAGE_GRAYSCALE);
    cv::Mat im2 = cv::imread(f2, CV_LOAD_IMAGE_GRAYSCALE);
    cv::Mat im3 = cv::imread(f3, CV_LOAD_IMAGE_GRAYSCALE);

    cv::CascadeClassifier faceCascade;
    bool isok = faceCascade.load("C:/test/detection/cv_bug/lbpcascade_frontalface.xml");
    //bool isok = faceCascade.load("C:/test/detection/cv_bug/haarcascade_frontalface_alt.xml");

    std::vector<cv::Rect> output1, output2, output3;
    faceCascade.detectMultiScale(im1, output1, 1.16, 1, 0, cv::Size(100, 100), cv::Size(0.99 * im1.cols, 0.99 * im1.cols));
    //faceCascade.detectMultiScale(im2, output2, 1.16, 1, 0, cv::Size(100, 100), cv::Size(0.99 * im2.cols, 0.99 * im2.cols));
    faceCascade.detectMultiScale(im3, output3, 1.16, 1, 0, cv::Size(100, 100), cv::Size(0.99 * im3.cols, 0.99 * im3.cols));

    std::cout << "02463d131.jpg: " << output1[0] << " Face detected: " << output1.size() << '\n';
    //std::cout << "02463d139.jpg: " << output2[0] << " Face detected: " << output2.size() << '\n';
    std::cout << "02463d142.jpg: " << output3[0] << " Face detected: " << output3.size() << '\n';

    std::cin.get();

    return 0;
}
```

![results](https://cloud.githubusercontent.com/assets/12760584/10251913/cb4b0822-693c-11e5-8069-73b80b4a62a9.png)

![02463d131](https://cloud.githubusercontent.com/assets/12760584/10251957/20319cca-693d-11e5-99db-577fe984ab96.jpg)
![02463d139](https://cloud.githubusercontent.com/assets/12760584/10251958/2182f970-693d-11e5-8789-77cb18740b11.jpg)
![02463d142](https://cloud.githubusercontent.com/assets/12760584/10251959/229b0e92-693d-11e5-8a2a-3f622835c761.jpg)

**Comment by almln** [comment]
This bug is connected with openCL. There is no problem with openCV build without openCL.

**Comment by StevenPuttemans** [comment]
OpenCL is known to cause issues. Either
- Explicitly disable OpenCL in OpenCV building process
- Or add `ocl::setUseOpenCL(false);` at the beginning of your code file

Both will make the sample work, just tested here locally.

**Comment by leugenea** [comment]
Tested again, now on Ubuntu 16.04.01, OpenCV 3.1.0, same issue but with HAAR features.

**Comment by StevenPuttemans** [comment]
@leugenea did you disable OpenCL?

**Comment by leugenea** [comment]
@StevenPuttemans , yes, without OpenCL everything works just fine. And even on PCs without GPU (Intel or NVidia).

**Comment by sturkmen72** [comment]
just a note:
i was tested this issue before and get same result like

![this image](https://cloud.githubusercontent.com/assets/12760584/10251913/cb4b0822-693c-11e5-8069-73b80b4a62a9.png)

i tested now and it works as expected.

( recently i updated display driver : Intel(R) HD Graphics 4000 10.18.10.4653 ) maybe outdated display driver was the reason. or the bug is fixed somehow.
